Makefile

```diff
@@ -34,7 +34,57 @@ endif
 GO=go
 GOFLAGS=""
 
+# requires:
+#	AWS_SECRET		- Secret from AWS
+#	AWS_ACCESS		- Access key from AWS
+#	S3_REGION		- Region of the bucket used to store the model
+#	S3_ENDPOINT		- Endpint of the bucket
+#	REGISTRY_USERNAME	- quay.io username
+#	REGISTRY_PASSWORD	- quay.io password
+#	REGISTRY_CREDENTIALS	- quay.io robot secret file
+go-test-setup:
+ifndef AWS_SECRET
+	$(error AWS_SECRET is undefined)
+endif
+ifndef AWS_ACCESS
+	$(error AWS_ACCESS is undefined)
+endif
+ifndef S3_REGION
+	$(error S3_REGION is undefined)
+endif
+ifndef S3_ENDPOINT
+	$(error S3_ENDPOINT is undefined)
+endif
+ifndef REGISTRY_USERNAME
+	$(error REGISTRY_USERNAME is undefined)
+endif
+ifndef REGISTRY_PASSWORD
+	$(error REGISTRY_PASSWORD is undefined)
+endif
+ifndef REGISTRY_CREDENTIALS
+	$(error REGISTRY_CREDENTIALS is undefined)
+endif
+	oc new-project test-namespace
+	@cat pipelines/tekton/aiedge-e2e/templates/credentials-s3.secret.yaml.template | sed 's#{{ AWS_SECRET }}#${AWS_SECRET}#g' | sed 's#{{ AWS_ACCESS }}#${AWS_ACCESS}#g' | sed 's#{{ S3_REGION }}#${S3_REGION}#g' | sed 's#{{ S3_ENDPOINT }}#${S3_ENDPOINT}#g' | oc create -f -
+	@cat pipelines/tekton/aiedge-e2e/templates/credentials-image-registry.yaml.template | sed 's#{{ REGISTRY_USERNAME }}#${REGISTRY_USERNAME}#g' | sed 's#{{ REGISTRY_PASSWORD }}#${REGISTRY_PASSWORD}#g' | oc create -f -
```

Comment:
I think we should simplify this entire setup workflow to have the user provide their own unique registry & credentials files instead of using the Makefile to process them.  

Both files are a known format and one time initialization and we only need the secretName for the PipelineRun execution.  This should simplify the workflow of having to setup the environment with secure information

Reply:
I did add this change but I am not sure on Landon's comment

Reply:
> I think we should simplify this entire setup workflow to have the user provide their own unique registry & credentials files instead of using the Makefile to process them.

@LaVLaS I think the intention of this Make target is just for automated test setup: it's not intended to be used by users. Are you suggesting that it would be simpler to have the provisioning of these secrets not be part of `go-test-setup`?

Reply:
Adding a comment here based on offline conversations, we will need to refactor this in follow-up PRs to separate the functionality into
1. Creation and application of the user objects that are based on the templates we provide (S3 credentials, image registry, test-data,...).  This should probably be a separate script that could be used for users and CI to ensure everything is processed and applied correctly to the server
2. Setup that is required specifically for the go-tests.

Right now we have divergent paths for the [shell tests](test/shell-pipeline-tests) and the functional tests proposed in this PR so we need to get to a state where we are utilizing the same workflow for e2e and functional tests

Reply:
@LaVLaS I think the approach is the same, we use the same templates for both go and shell tests. Just for Openshift CI the credentials need to be obtained from secrets.